webapp/src/main/webapp/src/app/sender/archive/state/sender-archive.reducer.ts

```diff
@@ -0,0 +1,47 @@
+import {createFeatureSelector, createSelector} from "@ngrx/store";
+import { SenderSyncArchiveCountAndItems } from "../sender-sync-archive-count-and-items";
+import { SenderArchiveAction, SenderArchiveActionType } from "./sender-archive.actions";
+
+export interface SenderArchiveState {
+	countAndItems?: SenderSyncArchiveCountAndItems;
+}
+
+const GET_SYNC_ARCHIVE_FEATURE_STATE = createFeatureSelector<SenderArchiveState>('senderArchiveQueue');
+
+export const GET_SYNC_ARCHIVE = createSelector(
+	GET_SYNC_ARCHIVE_FEATURE_STATE,
+	state => state.countAndItems
+);
+
+export const UPDATE_SYNC_ARCHIVE = createSelector(
+	GET_SYNC_ARCHIVE_FEATURE_STATE,
+	state => state.countAndItems
+);
+
+
+const initialState: SenderArchiveState = {
+	countAndItems: new SenderSyncArchiveCountAndItems()
+};
+
+export function senderArchiveReducer(state = initialState, action: SenderArchiveAction) {
+
+	switch (action.type) {
+
+		case SenderArchiveActionType.SENDER_ARCHIVED_LOADED:
+			return {
+				...state,
+				countAndItems: action.countAndItems
+			};
+		case SenderArchiveActionType.UPDATE_ARCHIVED_DATA:
```

Comment:
This action is basically a duplicate of SENDER_ARCHIVED_LOADED, you should instead have a different action to trigger a search and here you should instead be resetting(clearing) `countAndItems`

Reply:
Done